openshift-client/src/main/java/io/fabric8/openshift/client/DefaultOpenShiftClient.java

```diff
@@ -147,17 +146,22 @@ public DefaultOpenShiftClient(String masterUrl) throws KubernetesClientException
   protected DefaultOpenShiftClient(OkHttpClient httpClient, OpenShiftConfig config) throws KubernetesClientException {
     super(httpClient, config);
     try {
-      addOpenShiftAuthInterceptor();
-      this.delegate = new DefaultKubernetesClient(httpClient, config);
+      this.delegate = new DefaultKubernetesClient(clientWithOpenShiftOAuhtInterceptor(httpClient), config);
       this.openShiftUrl = new URL(config.getOpenShiftUrl());
     } catch (MalformedURLException e) {
       throw new KubernetesClientException("Could not create client", e);
     }
   }
 
-  private void addOpenShiftAuthInterceptor() {
-    httpClient = httpClient.newBuilder().authenticator(Authenticator.NONE).build();
-    httpClient = httpClient.newBuilder().addInterceptor(new OpenShiftOAuthInterceptor(httpClient,  OpenShiftConfig.wrap(getConfiguration()))).build();
```

Comment:
I'm confused. I've just run a test of swapping interceptors & don't get any problem with unmodifiable list:

```java
OkHttpClient client = HttpClientUtils.createHttpClient(new ConfigBuilder().build());
    System.out.println(client.interceptors());
    OkHttpClient.Builder builder = client.newBuilder();
    builder.interceptors().clear();
    System.out.println(builder.build().interceptors());
```